src/webui/www/private/views/search.html

```diff
@@ -462,36 +460,43 @@
             const populateCategorySelect = function(categories) {
                 const categoryHtml = [];
                 categories.each(function(category) {
-                    categoryHtml.push("<option>" + category + "</option>");
+                    const option = new Element("option");
+                    option.set("value", category.id);
+                    option.set("html", category.name);
+                    categoryHtml.push(option.outerHTML);
                 });
 
                 // first category is "All Categories"
-                if (categoryHtml.length > 1)
+                if (categoryHtml.length > 1) {
                     // add separator
-                    categoryHtml.splice(1, 0, "<option disabled>──────────</option>");
+                    const option = new Element("option");
+                    option.set("disabled", true);
+                    option.set("html", "──────────");
+                    categoryHtml.splice(1, 0, option.outerHTML);
+                }
 
                 $('categorySelect').set('html', categoryHtml.join(""));
             };
 
             const selectedPlugin = $('pluginsSelect').get("value");
+
             if ((selectedPlugin === "all") || (selectedPlugin === "enabled")) {
-                const url = new URI('api/v2/search/categories');
-                url.setData('name', selectedPlugin);
-                new Request.JSON({
-                    url: url,
-                    noCache: true,
-                    method: 'get',
-                    onSuccess: function(response) {
-                        populateCategorySelect(response);
-                    }
-                }).send();
+                const uniqueCategories = {};
```

Comment:
How about using a [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)? AFAIK it is the proper data type for our use case here.

Reply:
You're not wrong that a Map would work, but the Objects vs. Maps [section](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Objects_vs._Maps) doesn't highlight any differences that would benefit us. We also don't currently use Map anywhere, so it doesn't seem like much point to introducing it here.

Reply:
Reason I mentioned Map was because there was a security CVE on a library which I (indirectly) depends on. I recall the post mortem analysis (couldn't find the link) actually suggested that Map should be preferred over Object when using as a dictionary. I can't recall the report details, seems the attack was targeting object property.
And it kind of make sense to me as Map seems to be introduced in JS just for dictionary usage, it has a clearer intended audience than Object.
I must say I actually would favor Map over Object in all available cases, however I'll not focus too much on that in this PR.